SimPEG/data_misfit.py

```diff
@@ -154,43 +199,91 @@ def W(self, value):
         self._W = value
 
     def residual(self, m, f=None):
+        r"""Computes the data residual vector for a given model.
+
+        Where :math:`\mathbf{d}_obs` is the observed data vector and :math:`\mathbf{d}_{pred}`
+        is the predicted data vector for a model vector :math:`\mathbf{m}`, this function
+        computes the data residual:
+
+        .. math::
+            \mathbf{r} = \mathbf{d}_{pred} - \mathbf{d}_{obs}
+
+        Parameters
+        ----------
+        m : (n_param, ) numpy.ndarray
+            The model for which the function is evaluated.
+        f : None, SimPEG.fields.Fields (optional)
```

Comment:
Small change to follow [numpydocs style for optional parameters](https://numpydoc.readthedocs.io/en/latest/format.html#parameters)
```suggestion
        f : None or SimPEG.fields.Fields, optional
```